### ISSUE TITLE
Dapper.EntityFramework package setup

### ISSUE THREAD

**Issue by danda1man** [StackExchange/dapper-dot-net#570]
Is there something I need to do to use DbGeography types with the entity framework package?  I installed the package, but by queries keep returning an empty DbGeography object (all properties are null).

**Comment by mgravell** [comment]
Can you indicate what library version (dapper / dapper.ef) and target
framework (.net 4.5.2, .net core 1.0, etc) you are using, and I'll treble
check with an example on that framework.

**Comment by danda1man** [comment]
1.50, and my project is targetting 4.6.1

**Comment by mgravell** [comment]
First, you need to make sure you have the types installed locally - they are not installed by default, and are available from Microsoft here as "Microsoft® System CLR Types for Microsoft® SQL Server® 2012": https://www.microsoft.com/en-gb/download/details.aspx?id=29065 - note that there are separate installers for x86 and x64.

Next thing you need is the EF pieces of Dapper; there was a problem in the System.Runtime reference in 1.50.0, so I've fixed this and deployed as 1.50.2; so you need:

    install-package Dapper.EntityFramework

(or via https://www.nuget.org/packages/dapper.entityframework - not that strong-named versions are also available)

Next, you also need an assembly-binding redirect, because MS ships v11, but EF asks for v10. In your app.config / web.config, under `<configuration>` you can include:

```
  <runtime>
    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
      <dependentAssembly>
        <assemblyIdentity name="Microsoft.SqlServer.Types" publicKeyToken="89845dcd8080cc91" />
        <bindingRedirect oldVersion="10.0.0.0" newVersion="11.0.0.0" />
      </dependentAssembly>
    </assemblyBinding>
  </runtime>
```

And finally, you need to tell dapper "core" about EF:

    Dapper.EntityFramework.Handlers.Register();

With that all in place: it should work fine; sorry there are a few steps, but almost all of them are due to MS, not me!

string redmond = "POINT (122.1215 47.6740)";
            DbGeography point = DbGeography.PointFromText(redmond, DbGeography.DefaultCoordinateSystemId);
            DbGeography orig = point.Buffer(20);
            

            var fromDb = connection.QuerySingle<DbGeography>("declare @geos table(geo geography); insert @geos(geo) values(@val); select * from @geos",
                new { val = orig });

            Console.WriteLine($"Original area: {orig.Area}");
            Console.WriteLine($"From DB area: {fromDb.Area}");

Let me know how you get on!

**Comment by danda1man** [comment]
Working!  I worked around the Runtime by manually adding the facade before I tried installing it, and I already had the types installed with redirect, so it must have been the handler registrations.  For some reason I thought that was just magicked into the wrapper.  Thanks!
